### ISSUE TITLE
Great work, much faster than original KAN. However, the KAT_Group would fail if the input dimension cannot be divided by 8.

### ISSUE THREAD

**Issue by Eachen-Soong** [Adamdad/rational_kat_cu#16]
We can test this issue by modifying ```example/mnist.py```:

Here we set the KAT_Group initialized as identity.
When the hidden_dim is 256, on the first step of training, the x value remains the same after passing the KAT layer;
however, when the hidden_dim is changed from 256 to 255, only the first 248 dims remain the same!

```python
class NeuralNet(nn.Module):
    def __init__(self, activation_func):
        super(NeuralNet, self).__init__()
        self.fc1 = nn.Linear(784, 255)
        self.activation = activation_func
        self.fc2 = nn.Linear(255, 10)

    def forward(self, x):
        x = x.view(-1, 784)
        x = self.fc1(x)
        x1=x
        x = self.activation(x)
        res=x1-x
        print(torch.sum(torch.abs(res[:, :248])))
        print(torch.sum(torch.abs(res[:, 248:])))
        x = self.fc2(x)
        return x
...
if __name__ == "__main__":
    kat_activation = KAT_Group(mode='identity') # Placeholder for KAT_1DGroup if not accessible
    train_and_benchmark(kat_activation, 'KAT 1DGroup')
```

I understand that for large networks such as ViT, the hidden dim is pretty natural to be like 256, but if consider other datasets  such as time series, it's very possible that we need dimensions to be not 2^n.

We would definitely appreciate it if this issue could be fixed.

**Comment by Eachen-Soong** [comment]
Oh, I get it... The num_groups must be a divisor of hidden_dim
